containers/hawkular-metrics-index-truncator/src/main/java/org/hawkular/openshift/truncator/Truncator.java

```diff
@@ -0,0 +1,187 @@
+                // Get current replicas
+                replicas = openshiftClient.getReplicationControllerScale(this.metricsRC);
+                if (replicas != 0) {
+                    metricsReplicas = replicas;
+                }
+
+                replicas = openshiftClient.getReplicationControllerScale(this.heapsterRC);
+                if (replicas != 0) {
+                    heapsterReplicas = replicas;
+                }
+
+                // Scale to zero
+                openshiftClient.scaleReplicaitonController(this.heapsterRC, 0);
+                openshiftClient.scaleReplicaitonController(this.metricsRC, 0);
+
+                    // Truncate
+                truncator.execute();
+            } else {
+                log.info("Truncation was not necessary, skipping.");
+            }
+        } else {
+            throw new RuntimeException("No cassandra pods available, Is Hawkular Metrics running?");
+        }
+    }
+
+    private void logVersion() {
+        log.info("Hawkular Metrics Index Truncator v{}", VersionUtil.getVersion());
+    }
+
+    private void logTruncatorProperties() {
+        log.info("Configured truncator properties:\n" +
+                "\thawkular.metrics.rc-name = " + metricsRC + "\n" +
+                "\thawkular.metrics.heapster.rc-name = " + heapsterRC + "\n" +
+                "\thawkular.metrics.cassandra.pod-prefix = " + cassandraPodPrefix + "\n" +
+                "\thawkular.metrics.partition-threshold (bytes) = " + partitionThreshold + "\n" +
+                "\thawkular.metrics.cassandra.use-ssl = " + useSSL + "\n" +
+                "\thawkular.metrics.force-truncation = " + forceTruncation + "\n" +
+                "\thawkular.metrics.cassandra.keyspace = " + keyspace + "\n" +
+                "\thawkular.metrics.cassandra.connection.max-delay = " + cassandraConnectionMaxDelay + "\n" +
+                "\thawkular.metrics.cassandra.connection.max-retries = " + cassandraConnectionMaxRetries + "\n" +
+                "\thawkular.metrics.truncator.max-retries = " + truncatorMaxRetries +
+                "\thawkular.metrics.truncator.max-delay = " + truncatorMaxDelay);
+    }
+
+
+    public void run() {
+        this.logVersion();
+        this.logTruncatorProperties();
+        int retryCounter = 0;
+        while (retryCounter < truncatorMaxRetries) {
+            try {
+                this.truncate();
+                // success, restore replicas
+                this.restoreControllers();
+                return;
+            } catch (Exception e) {
+                log.error("truncation failed on retry " + retryCounter + " of " + truncatorMaxRetries);
+                retryCounter++;
+                if (retryCounter >= truncatorMaxRetries) {
+                    e.printStackTrace();
+                    return;
+                }
+            }
+        }
+        // failed after n attempts, no matter what happens we need to restore controller replicas.
```

Comment:
you have a `return` in the `if (retryCounter >= truncatorMaxRetries)` clause so this `restoreControllers` will not be reached